core/src/main/java/org/elasticsearch/cluster/node/DiscoveryNodes.java

```diff
@@ -468,6 +470,16 @@ public Delta emptyDelta() {
         return new Delta(null, null, localNodeId, DiscoveryNode.EMPTY_LIST, DiscoveryNode.EMPTY_LIST);
     }
 
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject("nodes");
```

Comment:
this is tricky - DiscoveryNodes carries more info then just the nodes - it also tells you which one is local and which one is master. If we do have a ToXContent implementation this should be represented - but you may not want to expose it everywhere. Since we only use it in one place now, maybe just roll it back and remove this?